### PR TITLE
AI Agent Built With LangChain and FireWorksAI

### PR DESCRIPTION
**Pull request by RichmondAlake**
Thank you for contributing to LangChain!

- **AI Agent Built With LangChain and FireWorksAI**: "community notebook"
    - **Description:** Added a new AI agent in the cookbook folder that integrates prompt compression using LLMLingua and arXiv retrieval tools. The agent is designed to optimize the efficiency and performance of research tasks by compressing lengthy prompts and retrieving relevant academic papers. The agent also makes uses of MongoDB to store conversational history and as it's knowledge base using MongoDB vector store
    - **Twitter handle:** https://x.com/richmondalake

### CODE REVIEW

**Review by baskaryan — COMMENTED** [review]
awesome! could we add these to https://github.com/langchain-ai/langchain/blob/master/cookbook/README.md

**Comment by RichmondAlake** [comment]
> awesome! could we add these to https://github.com/langchain-ai/langchain/blob/master/cookbook/README.md

Thanks. And added an entry.

File: cookbook/README.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ Example code for building applications with LangChain, with an emphasis on more
 
 Notebook | Description
 :- | :-
+[agent_fireworks_ai_langchain_mongodb.ipynb](https://github.com/langchain-ai/langchain/tree/master/cookbook/agent_fireworks_ai_langchain_mongodb.ipynb) | Build an AI Agent With Memory Using MongoDB, LangChain and FireWorksAI.
```
Comment (baskaryan) [inline cookbook/README.md:7]: could we add one for mongodb-langchain-cach... as well?

Comment (RichmondAlake) [reply cookbook/README.md:7]: Hi @baskaryan 

A use case for caching is currently in the works.
If we can get this merged into the repo, the team will look into getting a caching component incorporated into the notebook as soon as possible.

Thanks

Comment (RichmondAlake) [reply cookbook/README.md:7]: @baskaryan added the entry for cache as well

**Comment by SagarChhabriya** [comment]
https://github.com/langchain-ai/langchain/blob/master/cookbook/agent_fireworks_ai_langchain_mongodb.ipynb

image

**Pull request merged**